### PR TITLE
Routing, history & arch

### PR DESCRIPTION
**Pull request by balthazar**

### CODE REVIEW

File: test/routing.spec.ts [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+});
+
+test('The createRouter should manage root routes', t => {
+
+  const {routes} = createRouter(ROUTES);
+  t.is(routes.user({uid: '123'}), '/users/123');
+
+});
+
+test('The createRouter should manage child routes', t => {
```
Comment (balthazar) [inline test/routing.spec.ts]: createRouter should generate urls with child routes

File: test/routing.spec.ts [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+  const {routes} = createRouter(ROUTES);
+
+  t.is(routes.post({uid: 123, pid: 2}).toString(), '/users/123/2');
+  t.is(routes.post({uid: 123, pid: 2}).paragraph({paragraph: 1, word: 42}), '/users/123/2/paragraph/1/42');
+
+})
+
+test('LOCATION_CHANGE should affect the reducer state accordingly', t => {
```
Comment (balthazar) [inline test/routing.spec.ts]: router location change handler should update the pathname and routes

File: test/routing.spec.ts [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+  t.is(paragraphTwo.pathname, '/users/1/25/paragraph/2/1');
+  t.deepEqual(paragraphTwo.routes, [
+    {component: PostComponent, params: {uid: '1', pid: '25'}},
+    {component: ParagraphComponent, params: {paragraph: '2', word: '1'}}
+  ]);
+
+});
+
+test('HISTORY_PUSH should affect the reducer and create a task', t => {
```
Comment (balthazar) [inline test/routing.spec.ts]: update the pathname, routes and create a task

**Review by btford — APPROVED** [review]
LGTM

**Pull request merged**